### PR TITLE
feat(fctx): add context.Context-based request-scoped value access

### PR DESCRIPTION
**Pull request by marnickvda**
# Add `fctx` package for context.Context-based request-scoped values

## Description

New `fctx/` package that moves request-scoped values (logger, tracer, request ID) from `ServiceContext[T]` into standard `context.Context`. This enables gradual migration: services keep working with the old API while adopting `fctx` accessors at their own pace. Choosing `fctx` so the import package doesn't clash with the existing `context` package. 

### Why `context.Context`?

OpenTelemetry's tracing SDK already propagates spans through `context.Context`. By storing our logger, tracer, and request ID in the same place, we align with how the rest of the Go ecosystem works. The old `ServiceContext[T]` wrapper forced handlers to depend on echo.Context and couldn't carry trace context into downstream service calls without extra plumbing.

**What's new:**
- `fctx.From`, `fctx.Logger`, `fctx.MustLogger`, `fctx.Tracer`, `fctx.RequestID` accessors
- `fctx.Middleware` as the replacement for `ServiceContextMiddleware`
- `fctx.WithLogger` / `WithTracer` / `WithRequestID` setters
- `fastecho.BindValidate(ec, v)` top-level helper

**Interop:** The old `ServiceContextMiddleware` now also populates fctx keys, so consumers can use `fctx.Logger(ctx)` without switching middleware. `GetServiceContext` returns a best-effort shim when running under the new middleware.

**Deprecations:** `ServiceContext[T]`, `GetServiceContext`, `ServiceContextMiddleware`, `ServiceContext.BindValidate` are all marked `// Deprecated:`. Removal deferred to next major.

### Design influences

The API borrows from established Go patterns rather than inventing new ones:

- **stdlib `context.Context`**: request-scoped values in `context.Context`, not framework wrappers (same as `net/http`)
- **OpenTelemetry**: tracing already relies on `context.Context` for span propagation; keeping everything in the same context avoids a split between trace data and application data
- **chi `middleware.GetReqID`**: single-purpose accessors over a god struct
- **zap `NewNop()` / OTel `trace/noop`**: safe zero-value fallbacks instead of nil
- **Echo's request context bridge**: enrich `ec.Request().Context()` and write it back, rather than replacing echo.Context
- **stdlib `With*` pattern**: composable setters that return a new context

## Types of Changes

- New feature (non-breaking change which adds functionality)

## Tasks

- [x] `fctx/keys.go` with context key types and setters
- [x] `fctx/fctx.go` with public accessor API
- [x] `fctx/middleware.go` with new middleware
- [x] Unit tests for accessors and middleware (including edge cases)
- [x] Updated old middleware with fctx interop and deprecation comments
- [x] Compatibility tests for both migration directions
- [x] `fastecho.BindValidate` helper
- [x] Full verification (`go vet`, `go build`, `go test`)

## Review

- [x] Tests
- [x] Documentation
- [x] CHANGELOG

## Deployment Notes

No breaking changes. The old API continues to work unchanged.

### CODE REVIEW

**Review by AdrianCio — APPROVED** [review]

**Comment by AdrianCio** [comment]
double check the readme whether it needs some changes in examples

File: fctx/middleware.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			var fields []zap.Field
+
+			if spanCtx := trace.SpanContextFromContext(ctx); spanCtx.IsValid() {
+				fields = append(fields,
+					zap.String("trace_id", spanCtx.TraceID().String()),
+					zap.String("span_id", spanCtx.SpanID().String()),
+				)
+			}
+
+			if reqID := ec.Response().Header().Get(echo.HeaderXRequestID); reqID != "" {
```
Comment (thzois) [inline fctx/middleware.go:64]: See the other comments in other PRs around this.

**Comment by thzois** [comment]
I have a couple of comments before we merge this, connected also with the other PRs around `trace_id` and `request_id` and so on. 

However a few things that popup
- Do we really need to go through this migration? Can't we adapt existing `ServiceContext`? 
- If the answer to the previous questions is no, how do we need to change every user of `fastecho`?

**Comment by AdrianCio** [comment]
To me the move to context.Context makes the code cleaner because we don't spill echo outside the transport layer of the code (handler) and since it's by default supported by otel it seems like a much more clear approach. 

As for current usage, @marnickvda did a good job with deprecation so that no change is required and people can switch on the fly to the new way without being impacted

File: fctx/middleware.go (hunk 1 above)
Comment (marnickvda) [reply fctx/middleware.go:64]: I think the discussion around request_id and trace_id is outside of the scope of these changes. These changes are foundational, regardless of us choosing to use request_id, trace_id, or both.

**Comment by marnickvda** [comment]
> Do we really need to go through this migration? Can't we adapt existing ServiceContext?

Yes, ServiceContext is an abstraction on top of context.Context that doesn't need to exist. We can utilize context.Context and pass some helper methods (through `fctx`). It will simplify the fastecho, and also makes it easier to use since we don't depend on generics anymore (which we weren't using in our own services).

> If the answer to the previous questions is no, how do we need to change every user of fastecho?

I've marked the existing ServiceContext with `// Deprecated:` which will already tell people's IDE that an alternative approach should be used. Backward compatibility is kept for the moment, but I'd say it's removed from `v1.0.0` when we get there.

**Review by thzois — APPROVED** [review]
